lib/acyclicGraph.ml

```diff
@@ -173,7 +177,7 @@ module Make (Point:Point) = struct
                  (fun _ a ->
                     match a with
                     | Canonical n' ->
-                      n'.status <- NoMark;
+                      set_status n' NoMark;
```

Comment:
This seems incorrect, the status of `n` is supposed to be unchanged but with your PR it is shared with `n'`

Reply:
It is totally wrong, ephemerons only work on boxed values, while I mistakenly attached it to an unboxed field.

Reply:
I've pushed a fix, it should make more sense now.